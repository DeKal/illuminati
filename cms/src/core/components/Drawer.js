import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Drawer from '@material-ui/core/Drawer'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import { pages } from 'core/const/pages'

const LeftDrawer = ({ classes, open, history }) => (
  <Drawer
    className={classes.drawer}
    variant="persistent"
    anchor="left"
    open={open}
    classes={{
      paper: classes.drawerPaper
    }}
  >
    <div className={classes.drawerHeader}>Blogs</div>
    <NoPaddingList>
      {Object.keys(pages).map(key => {
        const page = pages[key]
        const isHighLight = isPathMatchUrl(history.location.pathname, page.url)
        return (
          <ListItem
            button
            key={key}
            data-test-id={`drawer-item-${key}`}
            className={
              isHighLight ? classes.highLightListItem : classes.listItem
            }
            onClick={() => history.push(page.url)}
          >
            <ListItemIcon>{page.icon}</ListItemIcon>
            <ListItemText primary={page.name} />
          </ListItem>
        )
      })}
    </NoPaddingList>
  </Drawer>
)

export default LeftDrawer

LeftDrawer.propTypes = {
  classes: PropTypes.object,
  open: PropTypes.bool,
  history: PropTypes.object,
  openDrawer: PropTypes.func,
  closeDrawer: PropTypes.func
}

const NoPaddingList = withStyles({
  padding: {
    padding: 0
  }
})(List)

const isPathMatchUrl = (pathName, url) => {
  if (url === '/' || pathName === '/') {
    return pathName === url
  }
  return pathName.includes(url)
}
