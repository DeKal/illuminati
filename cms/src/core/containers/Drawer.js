import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import Drawer from 'core/components/Drawer'

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex'
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  hide: {
    display: 'none'
  },
  drawer: {
    width: theme.drawerWidth,
    flexShrink: 0
  },
  drawerPaper: {
    width: theme.drawerWidth
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    justifyContent: 'center',
    textTransform: 'uppercase',
    fontFamily: theme.fonts.montSerrat,
    color: theme.colors.navTitle,
    backgroundColor: theme.colors.darkBlue,
    fontSize: theme.fontSizes.XL,
    lineHeight: theme.navBarHeight,
    fontWeight: 'bold',
    letterSpacing: '2px',
    boxShadow:
      '0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)',
    ...theme.mixins.toolbar
  },
  highLightListItem: {
    borderLeft: `6px ${theme.colors.highlight} solid`
  },
  listItem: {
    paddingLeft: '22px',
    '&:hover': {
      paddingLeft: '16px',
      borderLeft: `6px ${theme.colors.highlight} solid`
    }
  }
}))

const DrawerContainer = props => <Drawer classes={useStyles()} {...props} />
const DrawerWithRouter = withRouter(DrawerContainer)

const mapStateToProps = state => ({
  open: state.drawer.open
})

export default connect(mapStateToProps)(DrawerWithRouter)
