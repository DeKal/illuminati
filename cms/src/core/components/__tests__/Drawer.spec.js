import * as React from 'react'
import { shallow } from 'enzyme'
import Drawer from '../Drawer'

const classes = {
  drawer: '',
  drawerPaper: '',
  drawerHeader: ''
}
describe('Drawer', () => {
  it('should render Drawer when it opens', function() {
    const wrap = shallow(<Drawer classes={classes} open />)
    expect(wrap).toMatchSnapshot()
  })

  it('should render Drawer when it closes', function() {
    const wrap = shallow(<Drawer classes={classes} open={false} />)
    expect(wrap).toMatchSnapshot()
  })
})
