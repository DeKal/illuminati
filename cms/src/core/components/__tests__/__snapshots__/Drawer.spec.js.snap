// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should render Drawer when it closes 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className=""
  classes={
    Object {
      "paper": "",
    }
  }
  open={false}
  variant="persistent"
>
  <div
    className=""
  >
    Blogs
  </div>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Inbox"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Inbox"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Starred"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Starred"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Send email"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Send email"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Drafts"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Drafts"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="All mail"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="All mail"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Trash"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Trash"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Spam"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Spam"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Drawer))>
`;

exports[`Drawer should render Drawer when it opens 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className=""
  classes={
    Object {
      "paper": "",
    }
  }
  open={true}
  variant="persistent"
>
  <div
    className=""
  >
    Blogs
  </div>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Inbox"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Inbox"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Starred"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Starred"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Send email"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Send email"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Drafts"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Drafts"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="All mail"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="All mail"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Trash"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MailIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Trash"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      key="Spam"
    >
      <WithStyles(ForwardRef(ListItemIcon))>
        <MoveToInboxIcon />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        primary="Spam"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Drawer))>
`;
